server.go

```diff
@@ -222,22 +222,36 @@ func (s *Server) fsScanner(done <-chan void) {
 		log.Printf("Couldn't load hashcache.dat: %v", err)
 	}
 	fastPass := (cacheSeed != nil)
-	firstPassDone := false
+
+	// Additionally autosave the hash cache each minute
+	autosaveStop := false
```

Comment:
This'd be much easier with sync/atomic rather than mutexes btw. Or I guess channels would work well for this too.